lib/statsd/class-statsd.php

```diff
@@ -15,8 +15,25 @@ class StatsD {
 	 * @param string|array $stats The metric(s) to set.
 	 * @param float $time The elapsed time (ms) to log
 	 **/
-	public static function timing( $stats, $time ) {
-		self::update_stats( $stats, $time, 1, 'ms' );
+	public function timing( $stats, $time ) {
+		$this->update_stats( $stats, $time, 1, 'ms' );
+	}
+
+	/**
+	 * Tracks time of callback and then sets one or more timing values
+	 *
+	 * @param string|array $stats The metric(s) to set.
+	 * @param callback $callback to be tracked
+	 **/
+	public function time( $stats, $callback ) {
+		$start_time = microtime( true );
+
+		call_user_func( $callback );
```

Comment:
This won't pass any arguments to the function, so could only be on certain functions (or would cause errors).